### PR TITLE
adds define guard necessary to build on cygwin. adapted from #2066

### PR DESCRIPTION
**Pull request by ssfrr**
Also see [this comment](https://github.com/libuv/libuv/commit/c5593b51dc98715f7f32a919301b5801ebf1a8ce#r31834872)

@vtjnash

### CODE REVIEW

File: src/win/winapi.h [hunk 1]
```diff
@@ -4108,12 +4108,15 @@
 # define SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE 0x00000002
 #endif
 
+#if !defined(__UNICODE_STRING_DEFINED) && defined(__MINGW32__)
```
[the remark this answers is not in the document]
Comment (ssfrr) [reply src/win/winapi.h]: OK, I can remove the `__MINGW32__` check. I guess the line below would immediately fail otherwise.

**Comment by ssfrr** [comment]
I can definitely do this on `v1.x`, though I also need it on `master` to build and run the test I'm adding for #1960. Should I do 2 PRs?

File: src/win/winapi.h [hunk 2]
```diff
@@ -4108,12 +4108,15 @@
...
+#define __UNICODE_STRING_DEFINED
 /* from winternl.h */
 typedef struct _UNICODE_STRING {
   USHORT Length;
   USHORT MaximumLength;
   PWSTR  Buffer;
 } UNICODE_STRING, *PUNICODE_STRING;
+#endif // !__UNICODE_STRING_DEFINED && __MINGW32__
```
Comment (cjihrig) [inline src/win/winapi.h]: `/* */` for comments please.

**Comment by cjihrig** [comment]
> I can definitely do this on `v1.x`, though I also need it on `master` to build and run the test I'm adding for #1960. Should I do 2 PRs?

Anything that lands in v1.x will be merged over to the master branch as well.

**Comment by cjihrig** [comment]
@ssfrr are you planning to follow up on this? If not, can we close it out.

**Comment by ssfrr** [comment]
I've got it on my back-burner but probably won't get to it for a while. At some point I need to spin up my windows box and finish up this and also add the tests for #1960, but it's not going to happen soon. Feel free to close and I'll re-open if/when I have time to do some more windows stuff.

**Comment by cjihrig** [comment]
OK. Thanks. Just let us know when you're ready to resume this.

**Pull request closed without merging**